designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ActionService.scala

```diff
@@ -104,14 +118,13 @@ class ActionService(
     }
   }
 
-  class ActionProcessor[LatestScenarioDetailsShape: ScenarioShapeFetchStrategy](
-      extractVersionOnWhichActionIsDoneFromLatestScenarioDetails: ScenarioWithDetailsEntity[
-        LatestScenarioDetailsShape
-      ] => Option[VersionId]
+  class ActionProcessor[ScenarioDetailsShape: ScenarioShapeFetchStrategy](
+      extractVersionOnWhichActionIsDone: ScenarioWithDetailsEntity[ScenarioDetailsShape] => Option[VersionId],
+      scenarioGraphSource: ScenarioGraphSource = LatestVersion,
```

Comment:
Let's remove the default value for this parameter. The default value has sense when it is used very very often and is obvious. In any other situation, it can cause a potential bug because developer forgets to pass it.